### DISCUSSION TITLE
Sudoku App Concept

### DISCUSSION THREAD

**Discussion by Andres6936** [bryc/code#15] [Show and tell]
Hi @bryc , Thank you very much for the code for the Sudoku, it helped me as a starting point to create a small prototype application, it is not yet finished and I have many ideas to add and finish polishing the application, but I appreciate the fact of being able to have a record of the Sudokus that the user has been doing and also its solution.

I built the application in React + Vite, I leave a screenshot:

<p align="center">
<img alt="Sudoku App" src="https://user-images.githubusercontent.com/18540634/232066688-6f396f07-3263-4256-a6d1-8628d1af95eb.png" width="414"/>
</p>

###### Code in: https://github.com/Andres6936/Code/tree/master/sudoku

**Comment by bryc** [comment]
Hey there! Cool concept. Be aware my Sudoku code is pretty unfinished, I'm still learning things. The generator has no control over difficulty and might make puzzles too hard or too easy.

You can look at my [TODO List](https://github.com/bryc/code/blob/master/sudoku/TODO.md) for other resources in JS I have found.

**Reply by einsitang** [reply]
I have an algorithm idea about the sudoku generate, that is randomly dig holes with a complete Sudoku, but it needs to ensure the characteristics of its unique solution. Therefore, data verification is required for each digging process, and the difficulty of the generated sudoku depends on the number of holes dug.
here is projects for sudoku generator lib with different code.
- sudoku-nodejs: https://github.com/einsitang/sudoku-nodejs
- sudoku-go: https://github.com/einsitang/sudoku-go
- sudoku-dart: https://github.com/einsitang/sudoku-dart
